pymc3/tests/test_distributions.py

```diff
@@ -1684,10 +1684,15 @@ def test_lkj(self, x, eta, n, lp):
         decimals = select_by_precision(float64=6, float32=4)
         assert_almost_equal(model.fastlogp(pt), lp, decimal=decimals, err_msg=str(pt))
 
-    @pytest.mark.parametrize("n", [2, 3])
+    @pytest.mark.parametrize("n", [1, 2, 3])
     def test_dirichlet(self, n):
         self.pymc3_matches_scipy(Dirichlet, Simplex(n), {"a": Vector(Rplus, n)}, dirichlet_logpdf)
 
+    def test_dirichlet_with_unit_batch_shape(self):
+        with pm.Model() as model:
+            a = pm.Dirichlet("a", a=np.ones((1, 2)))
+        assert np.isfinite(model.check_test_point()[0])
```

Comment:
Now that I think about it, if the above `pymc3_matches_scipy`-based test doesn't test `Dirichlet.logp` under multiple/batched inputs, then we're probably not getting the coverage we need.

Something like the following would be more comprehensive:
```suggestion
    @pytest.mark.parametrize("a", [np.ones((1,)), np.ones((1, 2)), np.ones((2, 4, 3))])
    def test_dirichlet_with_batch_shapes(self, a):
        with pm.Model() as model:
            d = pm.Dirichlet("d", a=a)

        pymc3_res = d.distribution.logp(d.tag.test_value).eval()
        for idx in np.ndindex(a.shape[:-1]):
            scipy_res = stats.dirichlet(a[idx]).logpdf(d.tag.test_value[idx])
            assert np.array_equal(pymc3_res[idx], scipy_res)
```

Reply:
Good idea. I add this suggestion by using existing `dirichlet_logpdf` function.